internal/security/secretstore/kongadminapi.go

```diff
@@ -0,0 +1,241 @@
+//
+// The configuration file is a Kong declarative configuration file in YML format
+// that is written to an area where the Kong service has access and can import
+// the file.
+//
+// The JWT is saved to a JSON based file
+func (k *KongAdminAPI) Setup() error {
+
+	// Verify prefixes and paths
+	err := k.verifyPrefixes()
+	if err != nil {
+		return err
+	}
+
+	err = k.verifyPaths()
+	if err != nil {
+		return err
+	}
+
+	// Generate the keys for the Kong Admin API
+	key, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		return fmt.Errorf("%s Failed to generate private/public key pair: %w", k.prefixes.errText, err)
+	}
+
+	// Get an encoded version of the keys
+	k.secrets.private.encoded, _ = x509.MarshalECPrivateKey(key)
+	k.secrets.public.encoded, _ = x509.MarshalPKIXPublicKey(&key.PublicKey)
+
+	// PEM encode the keys
+	k.secrets.private.pem = string(pem.EncodeToMemory(&pem.Block{Type: "PRIVATE KEY", Bytes: k.secrets.private.encoded}))
+	k.secrets.public.pem = string(pem.EncodeToMemory(&pem.Block{Type: "PUBLIC KEY", Bytes: k.secrets.public.encoded}))
+
+	// When complete, clear the secrets portion of the KongAdminAPI struct
+	defer k.clearSecrets()
+
+	// Read in the configuration template
+	configTemplateBytes, err := ioutil.ReadFile(k.paths.template)
+	if err != nil {
+		return fmt.Errorf("%s Failed to read config template from file %s: %w", k.prefixes.errText, k.paths.template, err)
+	}
+
+	// Set random string for JWT issuer payload value
+	k.secrets.jwt.issuer = helper.GenerateRandomString(32)
+
+	// Insert public key
+	configTemplateText := strings.Replace(string(configTemplateBytes),
+		"<<INSERT-ADMIN-PUBLIC-KEY>>", strings.ReplaceAll(strings.TrimSpace(k.secrets.public.pem), "\n", "\n    "), -1)
+
+	// Insert issuer
+	configTemplateText = strings.Replace(configTemplateText,
+		"<<INSERT-ADMIN-JWT-ISSUER-KEY>>", k.secrets.jwt.issuer, -1)
+
+	// Write the config file to the configured save path
+	err = ioutil.WriteFile(k.paths.config, []byte(configTemplateText), 0644)
+	if err != nil {
+		return fmt.Errorf("%s Failed to write config template to file %s: %w", k.prefixes.errText, k.paths.config, err)
+	}
+
+	// Create the signed JWT
+	k.secrets.jwt.signed, err = k.createJWT(k.secrets.private.pem, k.secrets.jwt.issuer)
+	if err != nil {
+		return fmt.Errorf("%s Failed to create signed JSON Web Token: %w", k.prefixes.errText, err)
+	}
+
+	// Write JWT to secret file
+	err = ioutil.WriteFile(k.paths.jwt, []byte(k.secrets.jwt.signed), 0644)
+	if err != nil {
+		return fmt.Errorf("%s Failed to write JWT to file %s: %w", k.prefixes.errText, k.paths.jwt, err)
+	}
+
+	return nil
+}
+
+// createJWT creates a JWT from a given private key and issuer payload value
+func (k *KongAdminAPI) createJWT(privateKey string, issuer string) (signedToken string, err error) {
+
+	// Setup JWT generation variables
+	now := time.Now().Unix()
+	duration, err := time.ParseDuration("1h")
```

Comment:
Should the TTL value of `1h` be in the configuration or at least a constant?

Reply:
Doesn't matter to me - I simply copied over the code that was previously used in the service.go file that handled this.

Reply:
Seems like something a deployment may want to adjust depending on the environment.

Reply:
Ok - created a new configuration value to set this -> ConfigJWTDuration (configuration.toml)